gocd/gocd.go

```diff
@@ -206,6 +217,11 @@ func pipelineGET(gocdURL string, pipelineName string) (pipeline Pipeline, etag s
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode != 200 {
+		err = fmt.Errorf("Bad response code: %d", resp.StatusCode)
+		return pipeline, etag, err
+  }
+	
```

Comment:
Checking if the http code is not 200 is a great start.
More elaborate error checking can be done to get more info from GoCD like invalid pipeline structures, bad auth, and the like.
This will be extended in RHO-10